src/main/java/org/networkcalculus/dnc/network/server_graph/ServerGraph.java

```diff
@@ -749,6 +757,38 @@ public Flow getFlow(int id) throws Exception {
 		return result;
 	}
 
+	/**
+	 * Finds all the flows that have exactly path p as their path.
+	 *
+	 * @param p
+	 *             The path which all the returned flows have.
+	 * @return Set of flows that have p as their path.
+	 * @throws Exception
+	 *             If the path is null.
+	 */
+	public Set<Flow> getFlowsByPath(Path p) throws Exception {
+
+        // throw a runtime exception if p is null
+	    if(p == null)
+        {
+            throw new RuntimeException("Path argument for method getFlowsByPath(Path p) in ServerGraph was null.");
+        }
+
+	    else{
+            Set<Flow> result = new HashSet<Flow>();
+            // all the flows that start at first server of the path p are potential candidates
```

Comment:
Apply full sentence formatting here, too.
Shouldn't there be a definite article before "first"?